uint16 start_left
uint16 start_right
